auto_round/utils/device.py

```diff
@@ -732,6 +731,75 @@ def find_best_device(layer_name, estimated_memory, layer_idx):
     return ordered_device_map, names
 
 
+def get_moe_memory_ratio(block: torch.nn.Module) -> float:
+    """
+    Calculate the memory ratio for MoE (Mixture of Experts) models.
+
+    For MoE models, only num_experts_per_tok experts are activated per token,
+    not all experts. This function returns the ratio of active experts to total experts.
+
+    Args:
+        block (torch.nn.Module): The model block to analyze.
+
+    Returns:
+        float: Memory ratio (num_experts_per_tok / num_experts).
+               Returns 1.0 for non-MoE models.
+        bool: True if the model is MoE, False otherwise.
+
+    Examples:
+        - Non-MoE model: returns 1.0
+        - Mixtral (2/8 experts): returns 0.25
+        - Qwen2MoE (4/60 experts): returns ~0.067
+    """
+    from auto_round.utils.model import is_moe
+
+    for name, module in block.named_modules():
+        if is_moe(module):
```

Comment:
If not is_moe, continue. This way, all the remaining code will have fewer indentation levels, making it easier to read.